openfl/experimental/interface/fl_spec.py

```diff
@@ -0,0 +1,178 @@
+        # Submit flow to Runtime
+        self._metaflow_interface = MetaflowInterface(
+            self.__class__, self.runtime.backend
+        )
+        self._run_id = self._metaflow_interface.create_run()
+        if str(self._runtime) == "LocalRuntime":
+            # Setup any necessary ShardDescriptors through the LocalEnvoys
+            # Assume that first task always runs on the aggregator
+            self._setup_aggregator()
+            self._foreach_methods = []
+            FLSpec.reset_clones()
+            FLSpec.create_clones(self, self.runtime.collaborators)
+            # the start function can just be invoked locally
+            if self._checkpoint:
+                print(f"Created flow {self.__class__.__name__}")
+            try:
+                self.start()
+            except Exception as e:
+                if "cannot pickle" in str(e) or "Failed to unpickle" in str(e):
+                    msg = (
+                        "\nA serialization error was encountered that could not"
+                        "\nbe handled by the ray backend."
+                        "\nTry rerunning the flow without ray as follows:\n"
+                        "\nLocalRuntime(...,backend='single_process')\n"
+                        "\n or for more information about the original error,"
+                        "\nPlease see the official Ray documentation"
+                        "\nhttps://docs.ray.io/en/latest/ray-core/objects/serialization.html"
+                    )
+                    raise SerializationException(str(e) + msg)
+                else:
+                    raise e
+            for name, attr in final_attributes:
+                setattr(self, name, attr)
+        elif str(self._runtime) == "FederatedRuntime":
+            # Submit to director
+            raise Exception("Submission to remote runtime not available yet")
+        else:
+            raise Exception("Runtime not supported")
+
+    def _setup_aggregator(self):
+        """Sets aggregator private attributes as self attributes"""
+        for name, attr in self.runtime._aggregator.private_attributes.items():
+            setattr(self, name, attr)
+
+    @property
+    def runtime(self):
+        """Returns flow runtime"""
+        return self._runtime
+
+    @runtime.setter
+    def runtime(self, runtime):
+        """Sets flow runtime"""
+        if isinstance(runtime, Runtime):
+            self._runtime = runtime
+        else:
+            raise TypeError(f"{runtime} is not a valid OpenFL Runtime")
+
+    def capture_instance_snapshot(self, f, parent_func, kwargs):
+        """Takes backup of self before exclude or include filtering"""
+        return_objs = []
+        if "exclude" in kwargs:
+            exclude_backup = deepcopy(self)
+            return_objs.append(exclude_backup)
+        if "include" in kwargs:
+            include_backup = deepcopy(self)
+            return_objs.append(include_backup)
+        return return_objs
+
```

Comment:
Since we are not using `exclude_backup` and `include_backup` objects differently, could this be condensed to:

```
if "exclude" in kwargs or "include" in kwargs:
    return_objs = deepcopy(self)
return return_objs
```

Reply:
Fixed